Resources/config/menu.xml

```diff
@@ -13,6 +13,7 @@
         <service id="sonata.admin.menu.group_provider" class="Sonata\AdminBundle\Menu\Provider\GroupMenuProvider" public="false">
             <argument type="service" id="knp_menu.factory"/>
             <argument type="service" id="sonata.admin.pool"/>
+            <argument type="service" id="security.authorization_checker"/>
```

Comment:
Use the extension class to do a `replaceArgument` call to inject either the whole security context or just the authorization checker depending on the availability of the latter